### PR TITLE
chore(starter): Spring Boot 3.1 support

### PR DESCRIPTION
**Pull request by yanavasileva**
https://github.com/camunda/camunda-bpm-platform/issues/3444

### CODE REVIEW

**Comment by yanavasileva** [comment]
TODO engine-spring-6?: 

- [ ] https://github.com/camunda/camunda-bpm-examples/tree/master/deployment/spring-servlet-pa-wildfly
- [ ] https://github.com/camunda/camunda-bpm-examples/tree/master/deployment/embedded-spring-rest ?

File: spring-boot-starter/external-task-client/loan-granting-spring-boot-webapp/pom.xml [hunk 1]
```diff
@@ -85,13 +85,20 @@
       <artifactId>js-scriptengine</artifactId>
       <version>${graaljs.version}</version>
     </dependency>
+
+    <dependency>
+      <groupId>javax.xml.bind</groupId>
+      <artifactId>jaxb-api</artifactId>
+      <version>2.3.1</version>
```
Comment (psavidis) [inline spring-boot-starter/external-task-client/loan-granting-spring-boot-webapp/pom.xml:92]: 🔧 The version can be controlled from a property

**Comment by psavidis** [comment]
💭 Do you think it makes sense to use the `maven-enforcer-plugin` for the examples with min java version 17?

**Review by psavidis — APPROVED** [review]

**Comment by psavidis** [comment]
Great job, LGTM 👍

File: spring-boot-starter/external-task-client/loan-granting-spring-boot-webapp/pom.xml (hunk 1 above)
Comment (yanavasileva) [reply spring-boot-starter/external-task-client/loan-granting-spring-boot-webapp/pom.xml]: Done

**Comment by yanavasileva** [comment]
> Do you think it makes sense to use the maven-enforcer-plugin for the examples with min java version 17?

I adjusted the source and target configuration options to JDK 17, if the user uses lower version they will get an error:
`Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project loan-granting-spring-boot-webapp: Fatal error compiling: invalid target release: 17`
So far I don't see necessity to add an enforcer to the examples as we do for the implementation, we can always do this later.

**Pull request merged**